### ISSUE TITLE
Conch tests use 512-bit DSA keys

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7693]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Alex reported|
|-|-|
|Trac ID|trac#7693|
|Type|defect|
|Created|2014-10-24 23:35:43Z|
|Branch|https://github.com/twisted/twisted/tree/more-bits-7693|

`twisted.conch.test.keydata.publicDSA_openssh` is a 512-bit DSA key. This should be bumped up to be at-least 1024-bits.

512-bit keys are very very small, and factorable by an attacker. This would normally not matter, since it's just a test fixture, but pyca/cryptography will refuse to load a key this small as a result. And there is a desire to port conch to pyca/cryptography.

<details><summary>Searchable metadata</summary>

```
trac-id__7693 7693
type__defect defect
reporter__Alex Alex
priority__normal normal
milestone__ 
branch__branches_more_bits_7693 branches/more-bits-7693
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__None None
time__1414193743720587 1414193743720587
changetime__1446351949440438 1446351949440438
version__None None
owner__glyph glyph
cc__z3p cc__adiroiban
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

Why not update pyca to allow lower values when run in a 'insecure' mode ?

1024bit DSA is slow and it will slow the tests.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph commented|
|-|-|

I'd rather deal with the (small) performance hit to our tests than take the risk that people might actually use 512-bit keys for real.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

(In [46020]) Branching to more-bits-7693.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

Builders are spinning.

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|Alex set owner to @glyph|
|-|-|

once tests pass, lgtm

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_8"></a>|@glyph commented|
|-|-|

Thanks a bunch!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_9"></a>|@glyph set status to `closed`|
|-|-|

(In [46024]) Merge more-bits-7693: Use larger DSA keys in tests

Author: glyph

Reviewer: Alex

Fixes: [#7693](https://github.com/twisted/twisted/issues/7693)

Use 1024-bit DSA keys in unit tests.  This is a prerequisite for
switching Conch to use Cryptography, since Cryptography cannot load
512-bit DSA keys.
